### ISSUE TITLE
SectionList Container for Horizontal Usage

### ISSUE THREAD

**Issue by alzalabany** [facebook/react-native#18705]
- [y ] I have reviewed the [documentation](https://facebook.github.io/react-native)
- [ y] I have searched [existing issues](https://github.com/facebook/react-native/issues)
- [ y] I am using the [latest React Native version](https://github.com/facebook/react-native/releases)

## Steps to Reproduce
<SectionList 
    Horizontal
    data={ [ {title:'A',data:[1,2,3]}, {title:'B',data:[1,2,3]} ]} 
    renderHeader={({section})=><Text>{section.title}</Text>}
    renderItem={({item})=><Text>-{Item}-</Text>}
/>

## Expected Behavior
Horizontal make list flex-Direction=row, but this also affect items renderd, so we need to be able to wrap items of section with a Container, to show header on top of items not side by side.

## Actual Behavior
All headers and items will show in one row.

**Comment by roshangm1** [comment]
I tried to implement the solution provided by @kelset but I had problem having multiple flatlists rendered inside a section. For example, if I had two data to render, I will get two different flatlists and similarly, three flatlists with three data.
Current implementation: 
```
<SectionList
          sections={vendors}
          renderSectionHeader={({ section }) => (
            <Text> {section.category}</Text>
          )}
          renderItem={({ section }) => (
            <FlatList
              data={section.data}
              horizontal={true}
              renderItem={({ item }) => <Text>{item.name}</Text>}
            />
          )}
        />
```

**Comment by leleofg** [comment]
@roshangm1 Did you get it to work?

**Comment by roshangm1** [comment]
Yes. I got it work. I had to make changes in my data in order to accommodate this.

**Comment by c00piser** [comment]
@roshangm1 How did you do it ? currently I also have the same problem with multiple flatlist inside each section

**Comment by borgogelli** [comment]
Please @roshangm1 post an example

**Comment by AshokMadduru** [comment]
you can modify the sectionData to this:
```
sectionData = [
   {title: 'abc', data:[[{},{},{}]]},
   {title: 'abc', data:[[{},{},{}]]},
   .....
]
```
where each of the data array contain an array of objects inside it. in the rendering just mentioned the item
```
renderItem={({ item, index, section }) => (
<FlatList
  data={item}
   horizontal={true}
   renderItem={({ item }) => <Text>{item.name}</Text>}
   />
 )}
```

You can check this [here](http://ashokmadduru.blogspot.com/2019/03/horizontal-flatlist-inside-sectionlist.html)
